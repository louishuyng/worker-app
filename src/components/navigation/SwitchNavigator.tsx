import React from 'react';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { ThemeProvider } from 'styled-components';

import RootNavigator from './RootStackNavigator';
import { createTheme, ThemeType } from '../../utils/Theme';

const SwitchNavigator = createSwitchNavigator({
  RootNavigator,
}, {
  initialRouteName: 'RootNavigator',
});

const AppContainer = createAppContainer(SwitchNavigator);

export default function Navigator() {
  return (
    <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
      <AppContainer screenProps={{ theme: createTheme(ThemeType.LIGHT) }} />
    </ThemeProvider>
  );
}
