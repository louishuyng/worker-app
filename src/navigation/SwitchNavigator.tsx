import React from 'react';
import {
  createAppContainer,
  NavigationRouteConfigMap,
  createSwitchNavigator,
  SwitchNavigatorConfig,
} from 'react-navigation';
import { ThemeProvider } from 'styled-components';

import { createTheme, ThemeType } from 'utils/Theme';
import BottomMaterialNavigator from './BottomMaterialNavigator';

import { SignInScreen, SignUpStepOneScreen, SignUpStepTwoScreen } from 'screens/Auth';
import SendMailScreen from 'components/Password/RecoveryPassword/SendMail';
import { ConfirmMailScreen } from 'components/Password/RecoveryPassword';
import { ResetPasswordScreen } from 'components/Password';
import { RouteName } from 'constant';

const routeConfigMap: NavigationRouteConfigMap = {
  [RouteName.SIGN_IN]: SignInScreen,
  [RouteName.SIGN_UP_STEP_ONE]: SignUpStepOneScreen,
  [RouteName.SIGN_UP_STEP_TWO]: SignUpStepTwoScreen,
  [RouteName.SEND_MAIL]: SendMailScreen,
  [RouteName.CONFIRM_MAIL]: ConfirmMailScreen,
  [RouteName.RESET_PASSWORD]: ResetPasswordScreen,
  [RouteName.MAIN]: BottomMaterialNavigator,
};

const switchConfig: SwitchNavigatorConfig = {
  initialRouteName: 'SignUpStepOne',
};

const SwitchAnimatedNavigator = createSwitchNavigator(
  routeConfigMap,
  switchConfig
);

const AppContainer = createAppContainer(SwitchAnimatedNavigator);

export default function Navigator() {
  return (
    <ThemeProvider theme={createTheme(ThemeType.LIGHT)} >
      <AppContainer screenProps={{ theme: createTheme(ThemeType.LIGHT) }} />
    </ThemeProvider>
  );
}
