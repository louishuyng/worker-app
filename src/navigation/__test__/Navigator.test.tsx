import * as React from 'react';
import renderer from 'react-test-renderer';

import SwitchNavigator from '../SwitchNavigator';

jest.mock('NativeModules', () => ({
  PlatformConstants: {
    forceTouchAvailable: false,
  },
}));

describe('Switch Navigator', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<SwitchNavigator />);
    expect(component).toMatchSnapshot();
  });
});
