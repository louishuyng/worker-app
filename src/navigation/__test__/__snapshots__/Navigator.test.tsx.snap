// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Navigator renders without crashing 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#f6f6f6",
        },
      ]
    }
  >
    <View
      accessibilityElementsHidden={false}
      collapsable={false}
      importantForAccessibility="auto"
      pointerEvents="auto"
      removeClippedSubviews={false}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "top": 0,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "backgroundColor": "#eee",
                "flex": 1,
                "flexDirection": "column-reverse",
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "display": "flex",
                              "height": "100%",
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <Text>
                          Log In
                        </Text>
                      </RCTSafeAreaView>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "white",
                        }
                      }
                    >
                      <View
                        onLayout={[Function]}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "backgroundColor": "white",
                            "borderBottomColor": "#A7A7AA",
                            "borderBottomWidth": 0.5,
                            "height": 64,
                            "paddingBottom": 0,
                            "paddingLeft": 0,
                            "paddingRight": 0,
                            "paddingTop": 20,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "bottom": 0,
                                "flexDirection": "row",
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <View
                              pointerEvents="box-none"
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "bottom": 0,
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "left": 0,
                                  "opacity": 1,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                }
                              }
                            >
                              <Text
                                accessibilityTraits="header"
                                allowFontScaling={false}
                                numberOfLines={1}
                                onLayout={[Function]}
                                style={
                                  Object {
                                    "color": "black",
                                    "fontSize": 17,
                                    "fontWeight": "600",
                                    "marginHorizontal": 16,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Demo
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 0,
        "elevation": 8,
        "height": 133.4,
        "left": 0,
        "position": null,
        "right": 0,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 7,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 8,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "overflow": "hidden",
        }
      }
    >
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flexDirection": "row",
            "maxWidth": 384,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "backgroundColor": undefined,
              "borderRadius": 0,
              "height": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "top": 28,
              "transform": Array [
                Object {
                  "scale": 0.008,
                },
              ],
              "width": 0,
            }
          }
        />
        <View
          accessibilityLabel="JobList, tab, 1 of 4"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          clickable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "Helvetica Neue",
                      "fontSize": 8,
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityLabel="Calendar, tab, 2 of 4"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          clickable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "Helvetica Neue",
                      "fontSize": 8,
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityLabel="Notification, tab, 3 of 4"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          clickable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "Helvetica Neue",
                      "fontSize": 8,
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                }
              }
            />
          </View>
        </View>
        <View
          accessibilityLabel="User, tab, 4 of 4"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          clickable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "paddingVertical": 6,
            }
          }
        >
          <View
            pointerEvents="none"
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "height": 24,
                  "marginHorizontal": 12,
                  "marginTop": 2,
                  "transform": Array [
                    Object {
                      "translateY": 7,
                    },
                  ],
                  "width": 24,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Array [
                      Object {
                        "display": "flex",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "left": 0,
                      "position": "absolute",
                      "top": -2,
                    },
                    Object {
                      "right": -2,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "backgroundColor": "#f50057",
                      "borderRadius": 8,
                      "color": "#ffffff",
                      "fontFamily": "Helvetica Neue",
                      "fontSize": 8,
                      "height": 16,
                      "lineHeight": 16,
                      "minWidth": 16,
                      "opacity": 0,
                      "overflow": "hidden",
                      "paddingHorizontal": 4,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 16,
                  "paddingBottom": 2,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
